api/src/main/java/org/pmiops/workbench/exceptions/ExceptionUtils.java

```diff
@@ -46,24 +47,38 @@ public static RuntimeException convertGoogleIOException(IOException e) {
     throw new ServerErrorException(e);
   }
 
+  public static boolean isSocketTimeoutException(Exception e) {
+    return (e instanceof SocketTimeoutException);
+  }
+
+
   public static RuntimeException convertFirecloudException(ApiException e) {
     log.log(e.getCode() >= 500 ? Level.SEVERE : Level.INFO, "Exception calling FireCloud", e);
+    if (isSocketTimeoutException(e)) {
```

Comment:
Does this work? I thought `e.getCause()` was the timeout exception.

Reply:
Good catch. No, this won't work. Will fix shortly...